python/ray/tune/tests/test_trial.py

```diff
@@ -116,5 +117,52 @@ def test_trial_logdir_length():
     assert len(trial.storage.trial_dir_name) < 200
 
 
+def test_should_stop(caplog):
+    """Test whether `Trial.should_stop()` works as expected given a result dict."""
+    trial = Trial(
+        "MockTrainable",
+        stub=True,
+        trial_id="abcd1234",
+        stopping_criterion={"a": 10.0, "b/c": 20.0},
+    )
+
+    # Criterion is not reached yet -> don't stop.
+    result = _TrainingResult(
+        checkpoint=None, metrics={"a": 9.999, "b/c": 0.0, "some_other_key": True}
+    )
+    assert not trial.should_stop(result.metrics)
```

Comment:
Can we just pass in a dict without the `_TrainingResult` wrapper?

Reply:
done